File: fancycamera/src/main/java/co/fitcom/fancycamera/VideoEvent.java
/*
 * Created By Osei Fortune on 2/16/18 8:44 PM
 * Copyright (c) 2018
 * Last modified 2/16/18 7:39 PM
 *
 */

package co.fitcom.fancycamera;

import androidx.annotation.Nullable;

import java.io.File;

public class VideoEvent {
    private EventType mType;
    private File mFile;
    private String mMessage;
    VideoEvent(EventType type, @Nullable File file, @Nullable String message){
        mType = type;
        mFile = file;
        mMessage = message;
    }
    public EventType getType(){
        return mType;
    }
    public File getFile(){
        return  mFile;
    }

    public String getMessage(){
        return  mMessage;
    }

    public enum EventError{
        SERVER_DIED{
            @Override
            public String toString() {
                return "Server died";
            }
        },
        UNKNOWN{
            @Override
            public String toString() {
                return "Unknown";
            }
        }
    }

    public enum EventInfo{
        RECORDING_STARTED{
            @Override
            public String toString() {
                return "Recording started";
            }
        },
        RECORDING_FINISHED{
            @Override
            public String toString() {
                return "Recording finished";
            }
        },
        MAX_DURATION_REACHED {
            @Override
            public String toString() {
                return "Max duration reached";
            }
        },
        MAX_FILESIZE_APPROACHING{
            @Override
            public String toString() {
                return "Max filesize approaching";
            }
        },
        MAX_FILESIZE_REACHED{
            @Override
            public String toString() {
                return "Max filesize reached";
            }
        },
        NEXT_OUTPUT_FILE_STARTED{
            @Override
            public String toString() {
                return "Next output file started";
            }
        },
        UNKNOWN{
            @Override
            public String toString() {
                return "Unknown";
            }
        }
    }
}

File: fancycamera/src/main/java/co/fitcom/fancycamera/VideoEvent.kt
/*
 * Created By Osei Fortune on 2/16/18 8:44 PM
 * Copyright (c) 2018
 * Last modified 2/16/18 7:39 PM
 *
 */

package co.fitcom.fancycamera

import java.io.File

class VideoEvent internal constructor(val type: EventType, val file: File?, val message: String?) {

    enum class EventError {
        SERVER_DIED {
            override fun toString(): String {
                return "Server died"
            }
        },
        UNKNOWN {
            override fun toString(): String {
                return "Unknown"
            }
        }
    }

    enum class EventInfo {
        RECORDING_STARTED {
            override fun toString(): String {
                return "Recording started"
            }
        },
        RECORDING_FINISHED {
            override fun toString(): String {
                return "Recording finished"
            }
        },
        MAX_DURATION_REACHED {
            override fun toString(): String {
                return "Max duration reached"
            }
        },
        MAX_FILESIZE_APPROACHING {
            override fun toString(): String {
                return "Max filesize approaching"
            }
        },
        MAX_FILESIZE_REACHED {
            override fun toString(): String {
                return "Max filesize reached"
            }
        },
        NEXT_OUTPUT_FILE_STARTED {
            override fun toString(): String {
                return "Next output file started"
            }
        },
        UNKNOWN {
            override fun toString(): String {
                return "Unknown"
            }
        }
    }
}
